enumerables.rb

```diff
@@ -0,0 +1,114 @@
+module Enumerable
+  def my_each()
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    selected_array = []
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      selected_array << self[i] if yield(self[i])
+      i += 1
+    end
+    selected_array
+  end
+
+  def my_all?
+    i = 0
+    while i < size
+      return false if block_given? && !yield(self[i])
+      return false if self[i] == false || self[i].nil?
+
+      i += 1
+    end
+    true
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
```ruby
array = Array.new(100) { rand(0...9) }
array.my_all?(Integer) == array.all?(Integer)
```
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
```ruby
words = %w[dog door rod blade]
words.my_all?(/d/) == words.all?(/d/)
```
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
```ruby
array.my_all?(3) == array.all?(3) #true